builder/amazon/ebs/step_tag_ebs_volumes.go

```diff
@@ -17,37 +18,35 @@ func (s *stepTagEBSVolumes) Run(state multistep.StateBag) multistep.StepAction {
 	instance := state.Get("instance").(*ec2.Instance)
 	ui := state.Get("ui").(packer.Ui)
 
-	if len(s.VolumeRunTags) > 0 {
-		ui.Say("Tagging source EBS volumes...")
-
-		volumeIds := make([]*string, 0)
-		for _, v := range instance.BlockDeviceMappings {
-			if ebs := v.Ebs; ebs != nil {
-				volumeIds = append(volumeIds, ebs.VolumeId)
-			}
-		}
+	if len(s.VolumeRunTags) == 0 {
+		return multistep.ActionContinue
+	}
 
-		if len(volumeIds) == 0 {
-			return multistep.ActionContinue
+	volumeIds := make([]*string, 0)
+	for _, v := range instance.BlockDeviceMappings {
+		if ebs := v.Ebs; ebs != nil {
+			volumeIds = append(volumeIds, ebs.VolumeId)
 		}
+	}
 
-		tags := make([]*ec2.Tag, len(s.VolumeRunTags))
-		for key, value := range s.VolumeRunTags {
-			tags = append(tags, &ec2.Tag{Key: &key, Value: &value})
-		}
+	if len(volumeIds) == 0 {
+		return multistep.ActionContinue
+	}
 
-		_, err := ec2conn.CreateTags(&ec2.CreateTagsInput{
-			Resources: []*string{
-				instance.BlockDeviceMappings[0].Ebs.VolumeId,
-			},
-			Tags: tags,
-		})
-		if err != nil {
-			err := fmt.Errorf("Error tagging source EBS Volumes on %s: %s", *instance.InstanceId, err)
-			state.Put("error", err)
-			ui.Error(err.Error())
-			return multistep.ActionHalt
-		}
+	ui.Say(fmt.Sprintf("Adding tags to source EBS Volumes:"))
+	tags := common.ConvertToEC2Tags(s.VolumeRunTags, ui)
+
+	_, err := ec2conn.CreateTags(&ec2.CreateTagsInput{
+		Resources: []*string{
+			instance.BlockDeviceMappings[0].Ebs.VolumeId,
```

Comment:
I know this is existing behavior, but shouldn't we create a tag for each volume in the volumeIds list?